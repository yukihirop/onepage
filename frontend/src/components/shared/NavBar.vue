<template lang="pug">
b-navbar(fixed='top', toggleable='md', type='dark', variant='info')
  b-navbar-toggle(target='nav_collapse')
  b-navbar-brand(href='#') NavBar
  b-collapse#nav_collapse(is-nav='')
    b-navbar-nav
      b-nav-item(href='#') Link
      b-nav-item(href='#', disabled='') Disabled
    // Right aligned nav items
    b-navbar-nav.ml-auto
      b-nav-form
        b-form-input.mr-sm-2(size='sm', type='text', placeholder='Search')
          b-button.my-2.my-sm-0(size='sm', type='submit') Search
      b-nav-item-dropdown(text='Lang', right='')
        b-dropdown-item(href='#') EN
        b-dropdown-item(href='#') ES
        b-dropdown-item(href='#') RU
        b-dropdown-item(href='#') FA
      b-nav-item-dropdown(right='')
        // Using button-content slot
        template(slot='button-content')
          em User
        b-dropdown-item(href='#') Profile
        b-dropdown-item(href='#') Signout
</template>