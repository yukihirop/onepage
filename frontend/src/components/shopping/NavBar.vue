<template lang='pug'>
  nav.nav.has-shadow(style='box-shadow: 0 1px 0 rgba(219,219,219,.3);')
    .nav-left
      router-link.nav-item(to='/' )
        img(src='../../assets/logo.png', alt='Bulma logo')

    span.nav-toggle
      span
      span
      span

    .nav-right.nav-menu
      router-link.nav-item.is-tab(to='/',
      exact-active-class='is-active') Shop

      .nav-item.is-tab(:class="{'active-bottom-border': $route.path === '/cart'}")
        .field.is-grouped
          p.control
            router-link.button.is-info(to='/cart')
              span.icon
                i.fa.fa-shopping-cart
              span Checkout {{ itemsInCart }}
</template>

<script>
import { mapGetters } from 'vuex'
const namespace = 'shoppingModule'

export default {
  computed: {
    ...mapGetters(namespace, {
      products: 'cartProducts'
    }),
    itemsInCart () {
      let cart = this.$store.getters[namespace + '/cartProducts']
      return cart.reduce((accum, item) => accum + item.quantity, 0)
    }
  }
}
</script>

<style lang='stylus'>
.nav
  height auto
  margin-bottom 2rem

.nav-item
  img
    max-height 3.5rem

.active-bottom-border
  border-bottom 3px solid #00d1b2
  color #00d1b2
  padding-bottom calc(.75rem - 8px)
</style>
