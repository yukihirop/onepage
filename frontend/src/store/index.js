import Vue from 'vue'
import Vuex from 'vuex'
import shoppingModule from './modules/shopping'

Vue.use(Vuex)

const debug = process.env.NODE_ENV !== 'production'

export default new Vuex.Store({
  strict: debug,
  modules: {
    shoppingModule
  }
})