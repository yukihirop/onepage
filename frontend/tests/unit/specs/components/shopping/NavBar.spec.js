import Vuex from 'vuex'
import { shallow, createLocalVue } from 'vue-test-utils'
import NavBar from '@/components/shopping/NavBar'
import store from '@/store'
import factoryCart from '@specs/factories/components/shopping/Cart'

const localVue = createLocalVue()

localVue.use(Vuex)

const $route = {
  path: '/test'
}

describe('NavBar.vue', () => {
  it('初期状態のレンダリングのテスト', () => {
    const wrapper = shallow(NavBar, { store, localVue, mocks: { $route } })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })

  it('storeを用意してのレンダリングのテスト', () => {
    const shoppingModule = {
      namespaced: true,
      getters: {
        cartProducts: jest.fn().mockReturnValue(factoryCart.cartProducts)
      }
    }
    const mockedStore = {
      modules: {
        shoppingModule
      },
      mocks: {
        $route
      }
    }
    const store = new Vuex.Store(mockedStore)
    const wrapper = shallow(NavBar, { store, localVue })
    const template = wrapper.html()
    expect(template).toMatchSnapshot()
  })
})
