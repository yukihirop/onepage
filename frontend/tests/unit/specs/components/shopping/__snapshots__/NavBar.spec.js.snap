// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue storeを用意してのレンダリングのテスト 1`] = `
<nav class="nav has-shadow" style="box-shadow: 0 1px 0 rgba(219,219,219,.3);">
    <div class="nav-left">
        <router-link to="/" class="nav-item">
            <img src="../../assets/logo.png" alt="Bulma logo">
        </router-link>
    </div><span class="nav-toggle"><span></span><span></span><span></span></span>
    <div class="nav-right nav-menu">
        <router-link to="/" exact-active-class="is-active" class="nav-item is-tab">Shop</router-link>
        <div class="nav-item is-tab">
            <div class="field is-grouped">
                <p class="control">
                    <router-link to="/cart" class="button is-info"><span class="icon"><i class="fa fa-shopping-cart"></i></span><span>Checkout 60</span></router-link>
                </p>
            </div>
        </div>
    </div>
</nav>
`;

exports[`NavBar.vue 初期状態のレンダリングのテスト 1`] = `
<nav class="nav has-shadow" style="box-shadow: 0 1px 0 rgba(219,219,219,.3);">
    <div class="nav-left">
        <router-link to="/" class="nav-item">
            <img src="../../assets/logo.png" alt="Bulma logo">
        </router-link>
    </div><span class="nav-toggle"><span></span><span></span><span></span></span>
    <div class="nav-right nav-menu">
        <router-link to="/" exact-active-class="is-active" class="nav-item is-tab">Shop</router-link>
        <div class="nav-item is-tab">
            <div class="field is-grouped">
                <p class="control">
                    <router-link to="/cart" class="button is-info"><span class="icon"><i class="fa fa-shopping-cart"></i></span><span>Checkout 0</span></router-link>
                </p>
            </div>
        </div>
    </div>
</nav>
`;
